ai_oca_bridge_extra_parameters/models/ai_extra_parameter.py

```diff
@@ -0,0 +1,512 @@
+import ast
+import re
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.safe_eval import safe_eval
+
+
+class AIExtraParameter(models.Model):
+    _name = "ai.extra.parameter"
+    _description = "AI Bridge Extra Parameter"
+
+    name = fields.Char(required=True)
+    expression = fields.Char()
+    formula = fields.Text(
+        default="if object:\n" "    result = {'result_key': object.property}\n",
+    )
+    parameter_type = fields.Selection(
+        [
+            ("record", "Record"),
+            ("self", "Bridge Self"),
+        ],
+        required=True,
+        default="record",
+        help="Defines the context in which this parameter is evaluated. "
+        "'record' means it will be evaluated in the context of the record being processed, "
+        "'self' means it will be evaluated in the context of the AI bridge itself.",
+    )
+
+    evaluate_type = fields.Selection(
+        [
+            ("expression", "Expression"),
+            ("formula", "Formula"),
+        ],
+        required=True,
+        default="expression",
+        help="Defines how the parameter is evaluated. "
+        "'expression' means it will be evaluated as a Python expression, "
+        "'formula' means it will be evaluated as a formula expression.",
+    )
+
+    FORBIDDEN_PATTERNS = [
+        r"\.cr\b",
```

Comment:
This can be easily passed  by using something like ```getattr(self, "c"+"r")```or 

```
c = self.cr
e = cr.execute
e("SELECT *....")
```

Personally, I see a lot of problems on this approach.

Reply:
@etobella This point was finally resolved?

Reply:
With the forbidden patterns it should be fixed. 

In any case, this is (IMO) a double edge sword. It is interesting but it comes with  possible problems from a security perspective. Personally I will try to avoid it in my environments. I understand the reasons, but I don't like it. We added security rules in order to avoid this kind of issues, that is true, but I would prefer to avoid it.